packages/core/src/balance/Balance.utils.ts

```diff
@@ -33,16 +58,99 @@ export function convertToTxUnit(balance: BN, power: number): BN {
   return new BN(balance).mul(new BN(10).pow(new BN(15 + power)))
 }
 
-export function asFemtoKilt(balance: BN): BN {
-  return convertToTxUnit(balance, 0)
+export const TRANSACTION_FEE = convertToTxUnit(new BN(125), -9)
+
+/**
+ * Safely converts the given [[BalanceNumber]] to a string, using the supplied methods,
+ * or it given a string checks for valid number representation.
+ *
+ * @param input [[BalanceNumber]] to convert.
+ * @returns String representation of the given [[BalanceNumber]].
+ * @throws On invalid number representation if given a string.
+ * @throws On malformed input.
+ */
+export function balanceNumberToString(input: BalanceNumber): string {
+  if (typeof input === 'string') {
+    if (!input.match(/^-?\d*\.?\d+$/)) {
+      throw new Error('not a string representation of number')
+    }
+    return input
+  }
+  if (
+    typeof input === 'number' ||
+    (typeof input === 'object' &&
+      ((input instanceof BN && input.toString) ||
+        (input instanceof BigInt && input.toLocaleString)))
```

Comment:
Needs to be `typeof` to check for `bigint`

Reply:
now checks for typeof input === 'bigint' && input.toString